.circleci/config.yml

```diff
@@ -1395,91 +1396,276 @@ jobs:
+
+      - run:
+          name: build MSI package.
+          command: |
+             .\win32\build.ps1 c:\Python37
+              mkdir package
+              mv .\ScalyrAgentInstaller-*.msi package\ScalyrAgentInstaller.msi
+
+      - persist_to_workspace:
+          root: package
+          paths: ScalyrAgentInstaller.msi
+
+      - store_artifacts:
+          path: package
+
+  build-linux-packages:
+    description: "Build deb and rpm packages."
+    working_directory: ~/scalyr-agent-2
+    docker:
+        - image: circleci/python:3.6-jessie
+    steps:
+      - checkout
+      - restore_cache:
+          key: deps2-tox-{{ .Branch }}-3.6-venv-{{ checksum "dev-requirements.txt" }}
+
+      - run:
+          command: |
+            sudo apt-key update
+            sudo apt update
+            sudo apt install -y ruby ruby-dev rubygems rpm build-essential --force-yes
+
+            # TODO: looks like fpm upgraded its 'ffi' dependency version, which is now requres newer version of ruby.
+            # as a temporaty solution explicitly specify ffi version.
+            sudo gem install --no-document ffi:1.12.2 fpm
+
+      - run:
+          command: |
+            mkdir artifacts
+            mkdir package
+
+            ls
+            pushd artifacts
+
+            # update agent version to avoid version conflict with current scalyr repository package.
+            #echo "$(cat VERSION).1" > VERSION
+
+            python ../build_package.py deb
+            python ../build_package.py rpm
+
+            popd
+
+            cp artifacts/scalyr-agent-2*.deb package/scalyr-agent-2.deb
+            cp artifacts/scalyr-agent-2*.rpm package/scalyr-agent-2.rpm
+
+
+
+      - persist_to_workspace:
+          root: package
+          paths:
+            - scalyr-agent-2.deb
+            - scalyr-agent-2.rpm
+
+      - store_artifacts:
+          path: package
+
+
+
+  ami-tests:
+    description: "Run scalyr agent packages sanity tests based on EC2/Libcloud."
+    working_directory: ~/scalyr-agent-2
+    docker:
+      - image: circleci/python:3.6-jessie
+    steps:
+      - add_ssh_keys:
+            fingerprints:
+              - "7b:26:91:6b:a0:9d:31:50:4c:8d:31:a3:a6:b8:f2:7a"
+      - checkout
+      - attach_workspace:
+          at: /tmp/workspace
+      - run:
+          name: run package tests in EC2 instances.
+          command: |
+            export ACCESS_KEY=${AWS_ACCESS_KEY}
+            export SECRET_KEY=${AWS_SECRET_KEY}
+            export KEY_NAME=circleci
+            export PRIVATE_KEY_PATH="~/.ssh/id_rsa_7b26916ba09d31504c8d31a3a6b8f27a"
+            python -m pip install --user -r tests/ami/requirements.txt
+            export PYTHONPATH="."
+            mkdir -p outputs
+
+            # run AMI package sanity tests concurrently.
+            ./scripts/circleci/wait-for-running-bg-jobs.sh
+
+
+      - run:
+          name: ubuntu1804 result.
+          when: always
+          command: |
+            cat outputs/ubuntu1804.log
```

Comment:
You could run this as part of a single command to avoid a bunch of duplication :)

Reply:
But wouldn't it be too hard to navigate in this one big log file?

Reply:
Sorry, not a log file, but only one output frame in CircleCI.